From 96214998496a8984a4c7a9d798f9d31e43eacaf1 Mon Sep 17 00:00:00 2001
From: Suman Anna <s-anna@ti.com>
Date: Thu, 21 May 2015 10:42:16 -0500
Subject: [PATCH 15/18] remoteproc/pruss: lower the trace level in
 pru_rproc_mbox_callback

The current virtio rpmsg stack is such that it processes all the
pending messages in the vring on a single interrupt. So, it is
possible to see an empty interrupt without having to process any
messages, especially in cases where there is additional message
message communication that happens in-band within the callback
of a received message while processing it.

This behavior is seen when running the rpmsg_client_sample, and
printing the current trace message in pru_rproc_mbox_callback()
rather frequently, so lower the level of this trace from KERN_ERR
to KERN_DBG.

Reported-by: Jason Reeder <jreeder@ti.com>
Signed-off-by: Suman Anna <s-anna@ti.com>
---
 drivers/remoteproc/pruss_remoteproc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/remoteproc/pruss_remoteproc.c b/drivers/remoteproc/pruss_remoteproc.c
index 78d5806..b567b7a 100644
--- a/drivers/remoteproc/pruss_remoteproc.c
+++ b/drivers/remoteproc/pruss_remoteproc.c
@@ -589,7 +589,7 @@ static void pru_rproc_mbox_callback(struct mbox_client *client, void *data)
 
 	/* msg contains the index of the triggered vring */
 	if (rproc_vq_interrupt(pru->rproc, msg) == IRQ_NONE)
-		dev_err(dev, "no message was found in vqid %d\n", msg);
+		dev_dbg(dev, "no message was found in vqid %d\n", msg);
 }
 
 /* kick a virtqueue */
-- 
1.7.9.5

